### DISCUSSION TITLE
Support for Legacy Win7 Devices?

### DISCUSSION THREAD

**Discussion by madeff** [mitmproxy/mitmproxy#6049] [Q&A]
Hi.  I am looking to install mitmdump on several legacy 32-bit Win7 devices (cannot be upgraded because they are for medical devices -- not an issue because they are on isolated networks), but i cannot find any binaries for this.  Can anyone direct me to any older builds that include 32 bit windows?  Looking for mitmdump version 8 or earlier (as version 9 seems to treat all (non-TLS) traffic as web TLS (giving TLS errors) on our 64 bit win10 devices -- version 8 and earlier works fine.

**Comment by mhils** [comment]
You can find older releases at https://mitmproxy.org/downloads/, but please keep [our security policy](https://github.com/mitmproxy/mitmproxy/blob/main/SECURITY.md) in mind.

**Comment by madeff** [comment]
Hi Max,
Thanks for the reply.  I had already tried every version from 8.11 to 5.0
and they all gave the same error on 32 bit Win7:  So I assumed they were
all compiled as x64.  This is why I posted in Q&A.  I noticed that my post
disappeared -- did I do something wrong?
[image: image.png]

Marvin

**Reply by mhils** [reply]
>  I noticed that my post disappeared

Not at all. It looks like discussions that are marked as resolved are taken out of the main list... I did not know that. :-)

I unfortunately don't know when we dropped x86 binaries, it's been a while. You may be able to compile 32bit wheels for everything and then go that route (see https://github.com/mitmproxy/mitmproxy/discussions/6048), but it's probably quite a bit of hassle I would not recommend getting into.

**Comment by madeff** [comment]
You're right I'd rather avoid doing a whole process to compile back to
x86.  Any chance there are some old binaries laying around (old backups
maybe)?

**Reply by mhils** [reply]
I can only point you to https://mitmproxy.org/downloads/ unfortunately.

**Comment by PPeti66x** [comment]
The latest Win7 compatible version: 5.3.0. Versions beginning with 6.0 are dependent on "api-ms-win-core-path-l1-1-0.dll".

**Comment by Limour-dev** [comment]
Perhaps you can create a portable version of mitmproxy by imitating this.

https://github.com/Limour-dev/Python-Win7-x64/releases/tag/mitmproxy
